src/client/pythonEnvironments/base/locator.ts

```diff
@@ -86,10 +87,11 @@ export type BasicPythonLocatorQuery = {
  * emitted by watchers.
  *
  * @prop - searchLocations - if provided, results should be limited to
- *         within these locations
+ *         within these locations; `null` means "include envs that have
+ *         no search location"
  */
 export type PythonLocatorQuery = BasicPythonLocatorQuery & {
-    searchLocations?: Uri[];
+    searchLocations?: (Uri | null)[] | null;
```

Comment:
- What about when `searchLocations` is not defined, does it mean return all environments regardless of location? Doc comment would be useful
- Does `searchLocations` being `[null]` and `null` correspond to the same thing? In that case we should also include `Uri` in the type.
```suggestion
    searchLocations?: (Uri | null)[] | null | Uri;
```

Reply:
Here are the meanings:
* `null` - "ignore envs with a search location"
* `Uri[]` - "include all searched envs that match these locations, along with all non-searched envs"
* `(Uri | null)[]` - "include only envs that match these search locations"

It may be simpler to make `PythonLocatorQuery.searchLocations` be an object:

```suggestion
    searchLocations?: {
        roots: Uri[];
        includeNonRooted: boolean;
    };
```

Would that help clarify things?

Reply:
I'm going to make that change regardless. 😄

Reply:
Yeah I think that's better. Btw you didn't answer what does it mean if `searchLocations` is `undefined`?

Reply:
fixed

Reply:
> What about when searchLocations is not defined, does it mean return all environments regardless of location?

Yeah, it means ignore `env.searchLocation`.  I've clarified that in the doc comment.